src/python/gudhi/representations/vector_methods.py

```diff
@@ -885,3 +885,69 @@ def transform(self, X, sample_weight=None):
 
     def get_feature_names_out(self):
         return self._running_transform_names
+
+class PersistenceLengths(BaseEstimator, TransformerMixin):
+    """
+    This is a class that returns the sorted N-longest persistence lengths. If the input does not contain enough values,
+    the output will be filled with zeros.
+    """
+
+    def __init__(self, num_lengths=5):
+        """
+        Constructor for the PersistenceLengths class.
+
+        Parameters:
+            num_lengths (int): number of persistence lengths to return (default 5).
+
+        :raises ValueError: If num_lengths is lower or equal to 0.
+        """
+        if num_lengths <= 0:
+            raise ValueError("num_lengths must be greater than 0.")
+        self.num_lengths = num_lengths
+
+    def fit(self, X, y=None):
+        """
+        Fit the PersistenceLengths class on a list of persistence diagrams (this function actually does nothing but is
+        useful when PersistenceLengths is included in a scikit-learn Pipeline).
+
+        Parameters:
+            X (list of n x 2 or n x 1 numpy arrays): input persistence diagrams.
```

Comment:
Why nx1?

Reply:
Thanks, I missed this copy /paste that has to be changed. I fixed it on [e984496](https://github.com/GUDHI/gudhi-devel/pull/1117/commits/e984496218c0ee4b463a57e3ca6ceb2ffabe1945)